### PR TITLE
Add OpenStack inventory with the installer groups

### PR DESCRIPTION
**Pull request by tomassedovic**
This adds a second dynamic inventory which returns the groups expected
by the openshift-ansible roles. That is, it sets up `masters` and
`nodes` as expected.

This means it's not usable for scaling operations, but it should be a
drop-in replacement in all other situations.

Specifically, it should allow calling standalone opensthift-ansible
playbooks (e.g. custom logging setup) without having to wrap them in
another playbook that calls openstack/evaluate-groups.yml.

### CODE REVIEW

**Comment by tomassedovic** [comment]
/cc @tzumainn @mbruzek 

Okay I think we can get away without breaking anything.

So this adds a new inventory `playbooks/openstack/evaluate-groups.py` that sets the previous groups (e.g. `masters`, `nodes`, `glusterfs`) in addition to the "new ones".

Which means it should be a drop-in replacement for the 3.10 inventory, no `import_playbook: evaluate-groups.yml` required.

We still need an invenotory that *doesn't* set these values for the master scale up, but this should basically let us do both without code duplication and breakage.

If you're happy with this direction, I propose the following:

1. rename `inventory.py` to `scaleup-inventory.py`
2. rename the inventory in this patch  to `inventory.py`
3. update our docs for scaleup to use (with big bold text) `scaleup-inventory.py` when scaling.

I want you both to double-check this idea, but I think this means we don't actually need to break anything when moving from 3.10 to 3.11.

**Comment by tomassedovic** [comment]
pinging @akrzos as well

**Comment by mbruzek** [comment]
> If you're happy with this direction, I propose the following:  
> 1. rename inventory.py to scaleup-inventory.py
> 2. rename the inventory in this patch to inventory.py
> 3. update our docs for scaleup to use (with big bold text) scaleup-inventory.py when scaling.

@tomassedovic I am +1 for this idea, but I wanted to make a few more suggestions. Our team carries a [patch](https://github.com/redhat-performance/scale-ci-ansible/blob/master/roles/openshift_ansible_patch/files/30-inventory-no-app-nodes.patch) of inventory.py where we set an environment variable specifically for scaling events because getting the nodes each time takes longer as we approach high scale numbers like 2000 nodes. I have an old PR #7581 (which needs a rebase) that tries to get this in the product. If the new scaleup-inventory.py does not return all 2000 nodes each time, we would rather use inventory from upstream than carry a patch. However this is not what the current inventory.py does. Any possibility of making a scaleup-inventory.py that can achieve both goals?

**Comment by tomassedovic** [comment]
@mbruzek I dropped the ball on that completely, sorry!

Yes please rebase it I'll be happy to merge it afterwards. It does look more like a workaround than a real solution but if it's working at your scale, let's do that. I'd love to try & test a few options against your cluster for a better fix that doesn't drop nodes (if at all possible), but I don't want to block the immediate work on that.

Fwiw, the "scaleup inventory" would share the same code with inventory.py so a change there will affect both.

**Comment by mbruzek** [comment]
> please rebase it I'll be happy to merge it afterwards. It does look more like a workaround than a real solution but if it's working at your scale, let's do that
@tomassedovic  I have rebased https://github.com/openshift/openshift-ansible/pull/7581 please take another look.

**Comment by tomassedovic** [comment]
Okay, updated. This should remove the breaking change introduced earlier. The scaleup playbooks should now use `scaleup_inventory.py`, but everything that used to be working with `inventory.py` should continue to work unmodified.

It includes @mbruzek's patch that got merged in the meantime.

**Comment by mbruzek** [comment]
I am still +1 to this change, but when it lands it will require some changes to our automation:  
* Undo the [master group variable](https://github.com/redhat-performance/scale-ci-ansible/pull/173) change
* Start using the new `scaleup_inventory.py` in the scaleup role.

File: playbooks/openstack/configuration.md [hunk 1]
```diff
@@ -856,15 +859,15 @@ openshift_openstack_num_nodes: 8  # 5 existing and 3 new
...
+Next, run the appropriate playbook - either
 `openshift-ansible/playbooks/openstack/openshift-cluster/master-scaleup.yml`
 for master nodes or
 `openshift-ansible/playbooks/openstack/openshift-cluster/node-scaleup.yml`
 for other nodes. For example:
 
 ```
 $ ansible-playbook --user openshift \
-  -i openshift-ansible/playbooks/openstack/inventory.py \
+  -i openshift-ansible/playbooks/openstack/scaleup_inventory.py \
```
Comment (mbruzek) [inline playbooks/openstack/configuration.md]: Is there any other case where we should use the `scaleup_inventory.py` file? Such as `configure-new-nodes.yml` or `provision_resources.yml`?

Comment (tomassedovic) [reply playbooks/openstack/configuration.md]: If you don't rely on the `masters`, `etcd` etc. groups explicitly yourself, you can use the `scaleup_inventory` pretty much everywhere. All the playbooks under `openstack` should work with it. I mean, that's how it is now without this PR.

I've renamed it to let people use inventory.py as before (so they don't have to update their groups like you had to -- again sorry for not doing that in the first place).

And also for running the standalone non-openstack playbooks in this repo e.g. `playbooks/openshift-logging/config.yml` and similar.

**Comment by tomassedovic** [comment]
@tzumainn @luis5tb I can haz review?

**Comment by tomassedovic** [comment]
1. Yeah that's fair. I wanted to keep the interface out because it could potentially be different between the inventories, but it can't, really. Ansible's inventories already have a predefined interface and either way, we'd like to have these interchangeable.

2. evaluate_groups.yaml is still necessary for scaling up the master nodes. `scaleup_inventory.py` is basically the current `inventory.py` -- so it doesn't define `masters`, `nodes` etc. and therefore needs evaluate_groups everywhere you use it.

**Pull request merged**